src/spack_commands.py

```diff
@@ -26,14 +27,25 @@ def log_with_spack(command: str,
         spack_c2sm_path) / 'log' / machine_name() / test_category / filename
 
     # Setup spack env
-    spack_env = f'. {spack_c2sm_path}/setup-env.sh'
+    if uenv:
+        spack_env = f'. {spack_c2sm_path}/setup-env.sh /user-environment'
+    else:
+        spack_env = f'. {spack_c2sm_path}/setup-env.sh'
+
+    if uenv and srun:
+        uenv_args = f'--uenv={uenv}:/user-environment'
+    elif uenv and not srun:
+        uenv_args = f'squashfs-mount {uenv}:/user-environment/ -- '
+    else:
+        uenv_args = ''
 
     # Distribute work with 'srun'
-    if srun and getpass.getuser() == 'jenkins':
+    if srun:
```

Comment:
Please revert. I don't want to wait for the slurm queue if I run this test by hand.

Reply:
I disagree with some aspects of this line.

I found it confusing that we explicitly pass `srun` to function `log_with_spack` to just set it to `False` if not Jenkins user.
From my point of view local testing by us should not be different to what Jenkins does. It took my quite a while to realize that locally no compute node is used.
This matters, since i.e. ICON shows different behaviour at configure-stage when run on a compute node or not.

Furthermore on Daint you kill the login-node if you run `launch jenkins all` locally. There srun massively speeds up your builds.

Another downside of this line is, that I would need to introduce another boolean to compose the command for mounting the uenvs. It needs to be different if we srun or not.
```python
really_srun = (srun and getpass.getuser() == 'jenkins'):
if uenv and really_srun:
        uenv_args = f'--uenv={uenv}:/user-environment'
        ....
```
I prefer to explicitly handle this case a bit earlier over an one-liner that makes parts of the logic kind of obsolete.
But I see your point as well.

I have two suggestion:
1. Rename `srun`-arg in `log_with_spack` to `allow_srun` to indicate that srun is not used for all cases but possible.
2. Introduce an option/scope for the testsuite to switch on/off srun-command.


What do you think?

Reply:
`allow_srun` for now, `srun pytest ...` in the long run?

Reply:
`allow_srun` is not in place. Let's see what is suitable in the future.